src/precice/impl/ParticipantImpl.cpp

```diff
@@ -268,6 +268,10 @@ void ParticipantImpl::initialize()
 
   _meshLock.lockAll();
 
+  for (auto &context : _accessor->writeDataContexts()) {
+    PRECICE_CHECK(_accessor->meshContext(context.getMeshName()).provideMesh || !(_accessor->isDirectAccessAllowed(context.getMeshName()) && _allowsRemeshing), "Writing data via API access (configuration <write-data ... mesh=\"{}\") is not (yet) supported with remeshing", context.getMeshName());
+  }
+
```

Comment:
This check belongs into the `ParticipantConfiguration`, most likely into `finishParticipantConfiguration`. You can use `bool _remeshing` to check if remeshing is enabled.

Reply:
What's the problem here?

Reply:
What can be checked in the configuration should be checked there to raise an error as early as possible.
This also makes this checkable by `precice-tools check`.